src/test/java/com/segment/analytics/android/integrations/google/analytics/GoogleAnalyticsTest.java

```diff
@@ -195,6 +201,65 @@
     inOrder.verify(tracker).send(new HitBuilders.AppViewBuilder().setCustomMetric(14, 100).build());
   }
 
+  static AnalyticsContext contextWithCampaign(Campaign campaign) throws Exception {
+    Map<String, Object> map = new LinkedHashMap<>();
+    map.put("traits", new Traits());
+    Constructor<AnalyticsContext> constructor =
+        AnalyticsContext.class.getDeclaredConstructor(Map.class);
+    constructor.setAccessible(true);
+    AnalyticsContext analyticsContext = constructor.newInstance(map);
+    return analyticsContext.putCampaign(campaign);
+  }
+
+  @Test public void screenWithAllCampaignData() throws Exception {
+    Campaign campaign = new Campaign() //
+        .putContent("textlink") //
+        .putSource("google") //
+        .putMedium("cpc")
+        .putName("spring_sale");
+    ScreenPayload payload = new ScreenPayloadBuilder()  //
+        .name("hey") //
```

Comment:
For clarity, it's better to use to more realistic screen name than `hey`. If you want a placeholder, you can use `foo`/`bar`.